File: VB/WeCanFixIt/Models/IdentityModels.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity
Imports System.Security.Claims
Imports System.Threading.Tasks
Imports Microsoft.AspNet.Identity
Imports Microsoft.AspNet.Identity.EntityFramework
Imports Microsoft.AspNet.Identity.Owin

Namespace Models

    ' You can add profile data for the user by adding more properties to your ApplicationUser class, please visit https://go.microsoft.com/fwlink/?LinkID=317594 to learn more.
    Public Class ApplicationUser
        Inherits IdentityUser
        Public Async Function GenerateUserIdentityAsync(manager As UserManager(Of ApplicationUser)) As Task(Of ClaimsIdentity)
            ' Note the authenticationType must match the one defined in CookieAuthenticationOptions.AuthenticationType
            Dim userIdentity = Await manager.CreateIdentityAsync(Me, DefaultAuthenticationTypes.ApplicationCookie)
            ' Add custom user claims here
            Return userIdentity
        End Function
    End Class

    <Table("wo_worker")>
    Public Class Worker

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        <Column("wo_id")>
        Public Property Id As Int32

        <Required>
        <StringLength(50)>
        <Column("wo_name")>
        Public Property Name As String

        <Required>
        <StringLength(250)>
        <Column("wo_skills")>
        Public Property Skills As String

        <Required>
        <Column("wo_hours_worked")>
        <Range(1, 8)>
        Public Property HoursWorked As Nullable(Of Decimal)

        <Required>
        <Column("wo_rate")>
        <Range(1, 50)>
        Public Property Rate As Nullable(Of Int32)
    End Class


    Public Class ApplicationDbContext
        Inherits IdentityDbContext(Of ApplicationUser)
        Public Sub New()
            MyBase.New("DefaultConnection", throwIfV1Schema:=False)
        End Sub

        Public Property Workers As DbSet(Of Worker)

        'Public Property Purpose As String
        Public Shared Function Create() As ApplicationDbContext
            Return New ApplicationDbContext()
        End Function
    End Class
End Namespace

File: WeCanFixIt.Web/Models/IdentityModels.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;
using System.Security.Claims;
using System.Threading.Tasks;
using Microsoft.AspNet.Identity;
using Microsoft.AspNet.Identity.EntityFramework;

namespace WeCanFixIt.Web.Models
{
    // You can add profile data for the user by adding more properties to your ApplicationUser class, please visit http://go.microsoft.com/fwlink/?LinkID=317594 to learn more.
    public class ApplicationUser : IdentityUser
    {
        public async Task<ClaimsIdentity> GenerateUserIdentityAsync(UserManager<ApplicationUser> manager)
        {
            // Note the authenticationType must match the one defined in CookieAuthenticationOptions.AuthenticationType
            var userIdentity = await manager.CreateIdentityAsync(this, DefaultAuthenticationTypes.ApplicationCookie);
         
            // Add custom user claims here
            return userIdentity;
        }
    }

    [Table("wo_worker")]
    public class Worker
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("wo_id")]
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        [Column("wo_name")]
        public string Name { get; set; }

        [Required]
        [StringLength(250)]
        [Column("wo_skills")]
        public string Skills { get; set; }

        [Required]
        [Column("wo_hours_worked")]
        [Range(1,8)]
        public decimal? HoursWorked { get; set; }

        [Required]
        [Column("wo_rate")]
        [Range(1,50)]    
        public int? Rate { get; set; }
    }

    [Table("jo_job")]
    public class Job
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("jo_id")]
        public int Id { get; set; }

        [Column("jo_date",TypeName = "datetime2")]
        public DateTime Date { get; set; }

        [Required]
        [StringLength(50)]
        [Column("jo_work_type")]
        public string WorkType { get; set; }

        [Required]
        [Column("jo_amount")]
        public decimal Amount { get; set; }

        [Column("jo_hours")]
        public int? Hours { get; set; }

        [Column("jo_total")]
        public decimal? Total { get; set; }

    } 

    public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
    {
        public ApplicationDbContext()
            : base("DefaultConnection",false)
        {
        }

        public static ApplicationDbContext Create()
        {
            return new ApplicationDbContext();
        }

        public DbSet<Worker> Workers { get; set; }
        public DbSet<Job> Jobs { get; set; }

    }
}